examples/counters/src/components/Counter.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+import { enhanceComponent } from 'prism';
+
+import { EventHandler } from '../commonTypes';
+import { CounterState } from '../reducers/counterReducer';
+
+interface CounterPropsFromState {
+  value: number
+};
+
+interface CounterPropsFromDispatch {
+  onIncrement: EventHandler,
+  onDecrement: EventHandler
+};
+
+interface CounterProps extends CounterPropsFromState, CounterPropsFromDispatch {};
+
+const Counter = ({ onDecrement, onIncrement, value } : CounterProps) => (
+  <div>
+    <button onClick={onDecrement}>-</button>
+    <span>{value}</span>
+    <button onClick={onIncrement}>+</button>
+  </div>
+);
+
+const mapStateToProps = (state : CounterState) : CounterPropsFromState => ({
+  value: state.value
+});
+
+const mapDispatchToProps = (dispatch : Dispatch<CounterState>) : CounterPropsFromDispatch => ({
+  onIncrement: () => dispatch({ type: 'Increment' }),
+  onDecrement: () => dispatch({ type: 'Decrement' })
+});
+
+export default enhanceComponent<CounterState>(connect(
```

Comment:
I would suggest different approach here. I would move `enhanceComponent` to `CountersPair` in order to be able type general State. Now, you have to use `any`. `CountersPair` know the state `CounterState` anyway.